components/ManageAvailabilityWizard/AvailabilityModal.tsx

```diff
@@ -0,0 +1,313 @@
+          // part of time slot shifted to next day
+          result[day].push([newStartTime, "23:59"]);
+          if (nextDay != "sun") {
+            result[nextDay].push(["00:00", newEndTime]);
+          }
+        } else if (newEnd.day < endDateTime.day) {
+          // whole time slot was shifted to previous day
+          if (prevDay != "sun") {
+            result[prevDay].push([newStartTime, newEndTime]);
+          }
+        } else if (newEnd.day == endDateTime.day && newStart.day < startDateTime.day) {
+          // part of time slot shifted to previous day
+          result[day].push(["00:00", newEndTime]);
+          if (prevDay != "sun") {
+            result[prevDay].push([newStartTime, "23:59"]);
+          }
+        } else {
+          // no day shift
+          result[day].push([newStartTime, newEndTime]);
+        }
+      }
+    }
+    return result;
+  };
+
+  const updateAvailabilty = (day: string, times: string[][]): void => {
+    // function to update availability state when child component changes times
+    setAvailability((oldAvailability) => {
+      oldAvailability[day] = times;
+      return { ...oldAvailability };
+    });
+  };
+
+  // passes state values to callback and closes modal on save
+  const handleSubmit = async (): Promise<void> => {
+    setLoading(true);
+
+    try {
+      // converts empty availabilities to null
+      const newAvailability = {
+        mon:
+          !availability["mon"] || availability["mon"].length == 0
+            ? null
+            : sortTimes(availability["mon"]),
+        tue:
+          !availability["tue"] || availability["tue"].length == 0
+            ? null
+            : sortTimes(availability["tue"]),
+        wed:
+          !availability["wed"] || availability["wed"].length == 0
+            ? null
+            : sortTimes(availability["wed"]),
+        thu:
+          !availability["thu"] || availability["thu"].length == 0
+            ? null
+            : sortTimes(availability["thu"]),
+        fri:
+          !availability["fri"] || availability["fri"].length == 0
+            ? null
+            : sortTimes(availability["fri"]),
+        sat:
+          !availability["sat"] || availability["sat"].length == 0
+            ? null
+            : sortTimes(availability["sat"]),
+        timeZone: currTimeZone,
+      };
+
+      await client.updateAvailabilities(newAvailability, userId);
+
+      // update the cached data from useSWR
+      await mutate(newAvailability);
+    } catch (err) {
+      if (axios.isAxiosError(err) && err.response) setErrMsg(err.response.data);
+      else if (err instanceof Error) setErrMsg(err.message);
+      else setErrMsg("Error");
+      setLoading(false);
+      return;
+    }
+
+    setLoading(false);
+    handleClose();
+  };
+
+  if (currTimeZone != timezone) {
+    availabilityFromDB = convertToTimeZone(availabilityFromDB, currTimeZone, timezone);
```

Comment:
can we put this in a useEffect that runs once when the component is rendered?

Reply:
I tried moving that code into a useEffect when the component renders, but I ran into a too many rerender error. So I'm not too sure how to fix this.

Reply:
When does the error occur? Is it possibly because you're putting the statement into the same useEffect which depends on the availability state? Maybe try creating another useEffect that is dependent on nothing.